cmd/autoowners/main.go

```diff
@@ -133,40 +136,79 @@ func (orgRepo *orgRepo) getOwnersHTTP() (err error) {
 		data, err := gc.GetFile(orgRepo.Organization, orgRepo.Repository, filename, "HEAD")
 		if err != nil {
 			if _, nf := err.(*github.FileNotFound); nf {
+				logrus.WithField("orgRepo", fmt.Sprintf("%s/%s", orgRepo.Organization, orgRepo.Repository)).
+					WithField("filename", filename).
+					Debug("Not found file in the upstream repo")
+				orgRepo.OwnersFileExists = false
 				continue
 			} else {
 				return err
 			}
 		}
 
-		var target interface{}
+		orgRepo.OwnersFileExists = true
 		switch filename {
 		case "OWNERS":
-			target = &orgRepo.Owners
+			var target interface{}
+			target = &orgRepo.SimpleConfig
+			err = yaml.Unmarshal(data, target)
+			if err != nil {
+				logrus.WithField("data", string(data)).WithField("filename", filename).
+					WithField("orgRepo.Organization", orgRepo.Organization).
+					WithField("orgRepo.Repository", orgRepo.Repository).
+					WithError(err).Error("Unable to parse data for SimpleConfig.")
+				return err
+			}
+			if orgRepo.SimpleConfig.Empty() {
+				target = &orgRepo.FullConfig
+				err = yaml.Unmarshal(data, target)
+				if err != nil {
+					logrus.WithField("data", string(data)).WithField("filename", filename).
+						WithField("orgRepo.Organization", orgRepo.Organization).
+						WithField("orgRepo.Repository", orgRepo.Repository).
+						WithError(err).Error("Unable to parse data for FullConfig.")
+					return err
+				}
+				if len(orgRepo.FullConfig.Filters) == 0 {
+					return fmt.Errorf("invalid OWNERS file: %s", string(data))
+				}
+			}
 		case "OWNERS_ALIASES":
-			target = &orgRepo.Aliases
+			config := &struct {
+				Data map[string][]string `json:"aliases,omitempty"`
+			}{}
+			if err := yaml.Unmarshal(data, config); err != nil {
+				logrus.WithField("data", string(data)).WithField("filename", filename).
+					WithField("orgRepo.Organization", orgRepo.Organization).
+					WithField("orgRepo.Repository", orgRepo.Repository).
+					WithError(err).Error("Unable to parse data for RepoAliases.")
+				return err
+			}
+			result := RepoAliases{}
+			for alias, expanded := range config.Data {
+				result[github.NormLogin(alias)] = normLogins(expanded)
+			}
+			orgRepo.RepoAliases = result
 		default:
-			return fmt.Errorf("unrecognized filename %q", target)
+			return fmt.Errorf("unrecognized filename %s", filename)
 		}
-		err = yaml.Unmarshal(data, target)
-		if err != nil {
-			logrus.WithField("data", string(data)).WithField("filename", filename).
-				WithField("orgRepo.Organization", orgRepo.Organization).
-				WithField("orgRepo.Repository", orgRepo.Repository).
-				WithError(err).Error("Unable to parse data.")
-			return err
-		}
-	}
 
-	if orgRepo.Owners == nil && orgRepo.Aliases == nil {
-		return nil
 	}
-
 	orgRepo.Commit = sc.Commit.Tree.SHA
 	return nil
 
 }
 
+// copied from https://github.com/kubernetes/test-infra/blob/e1cbc3ccd91cdf2d1c229d61f996d12033a96500/prow/repoowners/repoowners.go#L546
+// TODO (hongkliu) make it public at upstream?
```

Comment:
Please :)

Reply:
Can I do it by another PR?
I want to get something working first.